beacon_node/network/src/sync/manager.rs

```diff
@@ -240,15 +265,107 @@ pub fn spawn<T: BeaconChainTypes>(
             log.clone(),
         ),
         range_sync: RangeSync::new(beacon_chain.clone(), log.clone()),
-        backfill_sync: BackFillSync::new(beacon_chain, network_globals, log.clone()),
-        block_lookups: BlockLookups::new(log.clone()),
+        backfill_sync: BackFillSync::new(beacon_chain.clone(), network_globals, log.clone()),
+        block_lookups: BlockLookups::new(
+            beacon_chain.data_availability_checker.clone(),
+            log.clone(),
+        ),
+        delayed_lookups: delayed_lookups_send,
         log: log.clone(),
     };
 
+    let log_clone = log.clone();
+    let sync_send_clone = sync_send.clone();
+    executor.spawn(
+        async move {
+            let slot_duration = beacon_chain.slot_clock.slot_duration();
+            // TODO(sean) think about what this should be
+            let delay = beacon_chain.slot_clock.unagg_attestation_production_delay();
+
+            loop {
+                let sleep_duration = match (
+                    beacon_chain.slot_clock.duration_to_next_slot(),
+                    beacon_chain.slot_clock.seconds_from_current_slot_start(),
+                ) {
+                    (Some(duration_to_next_slot), Some(seconds_from_current_slot_start)) => {
+                        if seconds_from_current_slot_start > delay {
+                            duration_to_next_slot + delay
+                        } else {
+                            delay - seconds_from_current_slot_start
+                        }
+                    }
+                    _ => {
+                        error!(log, "Failed to read slot clock");
+                        // If we can't read the slot clock, just wait another slot.
+                        sleep(slot_duration).await;
+                        continue;
+                    }
+                };
+
+                sleep(sleep_duration).await;
```

Comment:
This would sleep for 3 secs in which the `delayed_lookups_recv` won't receive anything right? Not sure if the intention is to sleep and then process all the messages on this channel at once.

Reply:
Yeah, the idea here was to process all the messages at once, which is why the while loop stops when it errors, this would hit `Err(TryRecvError::Empty)`. Do you think this is a reasonable way to do it? or do you think there's a better way?

Reply:
swapped it for an inteval here: 61b1a11e9104274a34e5e20df8ef3449e92fcc59